### PR TITLE
Fix Dart default ctor generation code around Options

### PR DESCRIPTION
**Pull request by Manishearth**

### CODE REVIEW

**Comment by Manishearth** [comment]
I'm going to preemptively merge this since Options are broken without it

File: tool/src/dart/mod.rs [hunk 1]
```diff
@@ -338,7 +338,15 @@ impl<'a, 'cx> TyGenContext<'a, 'cx> {
                 // If there's an existing zero-arg constructor, we repurpose it with optional arguments for all fields
                 let args = fields
                     .iter()
-                    .map(|field| format!("{}? {}", field.dart_type_name, field.name))
+                    .map(|field| {
+                        let question_mark = if field.ty.is_option() {
+                            // If it's optional we don't need to double-wrap in `?`
+                            ""
+                        } else {
+                            "?"
```
Comment (robertbastian) [inline tool/src/dart/mod.rs:346]: the formatter has `fmt_nullable` or something

File: feature_tests/dart/lib/src/OptionInputStruct.g.dart [hunk 2]
```diff
@@ -38,6 +36,15 @@ final class OptionInputStruct {
     return struct;
   }
 
+  factory OptionInputStruct({int? a, Rune? b, OptionEnum? c}) {
+    final result = _OptionInputStruct_default_ctor();
+    final dart = OptionInputStruct._fromFfi(result);
+    dart.a = a;
+    dart.b = b;
+    dart.c = c;
```
Comment (robertbastian) [inline feature_tests/dart/lib/src/OptionInputStruct.g.dart:44]: issue: these statements discard all fields from `default_ctor` and replace them with potential `null`s.

Comment (Manishearth) [reply feature_tests/dart/lib/src/OptionInputStruct.g.dart:44]: ah! Right. I'll fix this

**Pull request merged**